osquery/tables/system/hash.cpp

```diff
@@ -8,23 +8,54 @@
  *
  */
 
+#include <algorithm>
 #include <iomanip>
+#include <mutex>
+#include <set>
 #include <sstream>
+#include <unordered_map>
 #include <vector>
 
+#include <errno.h>
+#include <string.h>
+
+// clang-format off
+#include <sys/types.h>
+#include <sys/stat.h>
+// clang-format on
+
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
 #include <openssl/md5.h>
 #include <openssl/sha.h>
 
 #include <boost/filesystem.hpp>
 
+#include <osquery/core.h>
 #include <osquery/filesystem.h>
+#include <osquery/flags.h>
 #include <osquery/logger.h>
 #include <osquery/tables.h>
 
 #include "osquery/tables/system/hash.h"
 
 namespace osquery {
 
+FLAG(bool,
+     enable_persistent_hash_cache,
+     false,
+     "Cache calculated file hashes, re-calculate only if file has changed");
+FLAG(uint32,
+     persistent_hash_cache_max_size,
+     500,
+     "Cache hashes for upto this number of files");
+HIDDEN_FLAG(uint32,
```

Comment:
Can we just `#define` this? Do we need it as a flag for a specific reason?

Reply:
No specific reasons, just my previous experience with LRU caches saying the performance can be fine-tuned pretty well with this value. But I'm pretty sure it doesn't matter in this particular case, I don't expect a lot of evictions anyway. Made it a ```#define```.